logprep/runner.py

```diff
@@ -102,219 +75,130 @@ class Metrics(Component.Metrics):
-                self._config_refresh_interval = new_refresh_interval
-                self.metrics.config_refresh_interval += new_refresh_interval
-            self._schedule_config_refresh_job()
-            return
-        if refresh:
-            version_differ = new_configuration.get("version") != self._configuration.get("version")
-            if not version_differ:
-                self._logger.info(
-                    "Configuration version didn't change. Continue running with current version."
-                )
-                self._logger.info(
-                    f"Configuration version: {self._configuration.get('version', 'unset')}"
-                )
-                return
-        try:
-            new_configuration.verify(self._logger)
-
-            # Only reached when configuration is verified successfully
-            self._configuration = new_configuration
+            self.metrics.number_of_config_refresh_failures += 1
+            self._config_refresh_interval = int(self._config_refresh_interval / 4)
             self._schedule_config_refresh_job()
-            self._manager.set_configuration(self._configuration)
-            self._manager.restart()
-            self._logger.info("Successfully reloaded configuration")
-            config_version = self._configuration.get("version", "unset")
-            self._logger.info(f"Configuration version: {config_version}")
-            self.metrics.version_info.add_with_labels(1, self._metric_labels)
-            self.metrics.number_of_config_refreshes += 1
-            if self._config_refresh_interval is not None:
-                self.metrics.config_refresh_interval += self._config_refresh_interval
+        except ConfigVersionDidNotChangeError as error:
+            self._logger.info(str(error))
         except InvalidConfigurationError as error:
-            self._logger.error(
-                "Invalid configuration, leaving old"
-                f" configuration in place: {self._yaml_path}: {str(error)}"
-            )
+            self._logger.error(str(error))
+            self.metrics.number_of_config_refresh_failures += 1
+
+    def _set_logprep_version_info(self):
+        self.metrics.version_info.add_with_labels(
+            1,
+            {"logprep": f"{get_versions()['version']}", "config": self._configuration.version},
+        )
+
+    def stop(self):
+        """Stop the current process"""
+        self._exit_received = True
 
     def _schedule_config_refresh_job(self):
         refresh_interval = self._config_refresh_interval
         scheduler = self.scheduler
         if scheduler.jobs:
             scheduler.cancel_job(scheduler.jobs[0])
         if isinstance(refresh_interval, (float, int)):
-            refresh_interval = 5 if refresh_interval < 5 else refresh_interval
-            scheduler.every(refresh_interval).seconds.do(self.reload_configuration, refresh=True)
+            self.metrics.config_refresh_interval += refresh_interval
```

Comment:
it is not possible because of line 196 `refresh_interval` can't be None. For other cases if we call `add_with_labels` with a None labels dict or a None value I added tests and they raise TypeError for not supportet operations

Reply:
Ah true, we missed that `isinstance` check in the review. :+1: